src/H5Eint.c

```diff
@@ -734,6 +1478,63 @@ H5E__push_stack(H5E_t *estack, const char *file, const char *func, unsigned line
     FUNC_LEAVE_NOAPI(ret_value)
 } /* end H5E__push_stack() */
 
+/*-------------------------------------------------------------------------
+ * Function:    H5E__copy_stack_entry
+ *
+ * Purpose:     Copy a stack entry
+ *
+ * Return:      SUCCEED/FAIL
+ *
+ *-------------------------------------------------------------------------
+ */
+static herr_t
+H5E__copy_stack_entry(H5E_entry_t *dst_entry, const H5E_entry_t *src_entry)
+{
+    herr_t ret_value = SUCCEED; /* Return value */
+
+    /*
+     * WARNING: We cannot call HERROR() from within this function or else we
+     *		could enter infinite recursion.  Furthermore, we also cannot
+     *		call any other HDF5 macro or function which might call
+     *		HERROR().  HERROR() is called by HRETURN_ERROR() which could
+     *		be called by FUNC_ENTER().
+     */
```

Comment:
Is the comment above correct? If so, isn't the usage of `H5I_inc_ref` below problematic in the same way it was when the `H5I_inc_ref_noherr` function was added? That is, if `H5I_inc_ref` fails, it will use the `HGOTO_ERROR` macro.

Reply:
> Is the comment above correct? If so, isn't the usage of `H5I_inc_ref` below problematic in the same way it was when the `H5I_inc_ref_noherr` function was added? That is, if `H5I_inc_ref` fails, it will use the `HGOTO_ERROR` macro internally.

Ah, that's out of date, from the original routine I copied when creating this routine.  I'll remove the comment and switch the 'HGOTO_DONE' to HGOTO_ERROR'.

Reply:
> > Is the comment above correct? If so, isn't the usage of `H5I_inc_ref` below problematic in the same way it was when the `H5I_inc_ref_noherr` function was added? That is, if `H5I_inc_ref` fails, it will use the `HGOTO_ERROR` macro internally.
> 
> Ah, that's out of date, from the original routine I copied when creating this routine. I'll remove the comment and switch the 'HGOTO_DONE' to HGOTO_ERROR'.

Updated